src/NuGet/Microsoft.NETCore.Compilers.nuspec

```diff
@@ -26,13 +26,12 @@
     <file src="$additionalFilesPath$/Microsoft.NETCore.Compilers.props" target="build" />
 
     <!-- Targets and task files -->
-    <file src="Dlls/MSBuildTask/netstandard1.3/publish/Microsoft.Build.Tasks.CodeAnalysis.dll" target="tools" />
-    <file src="Dlls/MSbuildTask/netstandard1.3/publish/Microsoft.CSharp.Core.targets" target="tools" />
-    <file src="Dlls/MSbuildTask/netstandard1.3/publish/Microsoft.VisualBasic.Core.targets" target="tools" />
+    <file src="Dlls/MSBuildTask/netcoreapp2.0/Microsoft.Build.Tasks.CodeAnalysis.dll" target="tools" />
+    <file src="Dlls/MSbuildTask/netcoreapp2.0/Microsoft.CSharp.Core.targets" target="tools" />
+    <file src="Dlls/MSbuildTask/netcoreapp2.0/Microsoft.VisualBasic.Core.targets" target="tools" />
     <!-- N.B.: The backslashes below cannot be replaced with forward slashes. 
          https://github.com/NuGet/Home/issues/3584 -->
-    <file src="Dlls\MSBuildTask\netstandard1.3\publish\System.*.dll" target="tools" />
```

Comment:
What's the logic behind the removal of the System assemblies? This is meant to cover the cases where we bring in dependencies (like S.R.M or S.C.I) that are newer than the versions available in the chosen CLI.

Reply:
The reason is that they are simply no longer present. Now that we've properly multi-targetted this task the single and only DLL present after publish is the pipes DLL.

Reply:
Isn't that just luck? That is, the version of SRM/SCI that we reference is present in netcoreapp2.0. But if we upgrade to a newer one, shouldn't the new reference assembly be copied out? Otherwise I don't see how it could work properly.

Reply:
I agree it won't copy those if we change them. But I don't see that as a negative here. It forces us to be explicit about the dependencies we take here. 

Note: the uses of System.* was IMHO equally as wrong. Not every facade begins with System. For example Microsoft.Win32. Same argument could be made for that as S.C.I.

Reply:
Isn't the point of publish that determining the runtime dependencies via inspection can be intractable? It seems like if we go this route we should also have some way of failing the build if extra files do get generated, to make sure we don't drop them on the floor, only for the compiler to fail at runtime.

Reply:
As I see it we have two options:

1. Take everything publish puts out and just trust that it's right.
2. Hard failure on anything unexpected being introduced and require a hand-audit for each new assembly.

How it is now seems like the worst of all worlds.

Reply:
> Take everything publish puts out and just trust that it's right

Our loading logic often depends on us probing for DLLs based on their name and version. How does blindly copying everything to the output directory help for this? It just means the DLL will be there, not that we can load it.

Reply:
I think the loading logic for CoreCLR always prefers DLL loads next to the requesting assembly. This is why we only have redirect logic for desktop and for the runtimes/ directory.

Reply:
> I think the loading logic for CoreCLR always prefers DLL loads next to the requesting assembly

This is true for applications. In this case we are a plugin to an application. Does the logic still hold there? 

> This is why we only have redirect logic for desktop and for the runtimes/ directory. 

What is the recommendation for the desktop case then: blindly publish or not? If it's blindly publish then how do you reconcile that with the idea that we may have to probe for versions?

Reply:
What do you think the proper entry is here?

Reply:
Here are the entries I considered and why none of them work:

- `netcoreapp2.0/publish/**`: unnecessarily brings in PDB and XML files 
- `netcoreapp2.0/publish/System.*`: works for only part of the facades out there. 
- `netcoreapp2.0/publish/System.*` + `microsoft.*`: this double writes our own binaries in the output folder 

> Take everything publish puts out and just trust that it's right.

The problem is that I'm looking at the output and it doesn't seem right. The publish directory contains items we don't want (at least that I can see).

Reply:
OK, you've convinced me that (1) isn't really an option either. 😄 How should we handle the case that we upgrade assemblies and new stuff needs to be published?

Reply:
I think the only thing we can do is manual update + prioritize bootstrapping out of our NuPkg files. That provides some automated checks that we have the right items there. 

https://github.com/dotnet/roslyn/issues/24456